gnomad/sample_qc/relatedness.py

```diff
@@ -987,24 +985,29 @@ def maximal_independent_set_keep_samples(
     return related_samples_to_drop_ht
 
 
-def filter_mt_to_trios(mt: hl.MatrixTable, fam_ht: hl.Table) -> hl.MatrixTable:
+def filter_mt_to_trios(
+    t: Union[hl.vds, hl.matrixtable], fam_ht: hl.Table
+) -> Union[hl.vds, hl.matrixtable]:
     """
-    Filter a MatrixTable to a set of trios in `fam_ht` and annotates with adj.
+    Filter a Matrix Table or a Variant Dataset to a set of trios in `fam_ht`.
 
-    :param mt: A Matrix Table to filter to only trios
+    :param t: A Variant Dataset to filter to only trios
     :param fam_ht: A Table of trios to filter to, loaded using `hl.import_fam`
-    :return: A MT filtered to trios and adj annotated
+    :return: A Matrix Table or a Variant Dataset with only the trios in `fam_ht`
     """
-    # Filter MT to samples present in any of the trios
+    # Filter VDS to samples present in any of the trios
     fam_ht = fam_ht.annotate(fam_members=[fam_ht.id, fam_ht.pat_id, fam_ht.mat_id])
     fam_ht = fam_ht.explode("fam_members", name="s")
     fam_ht = fam_ht.key_by("s").select().distinct()
 
-    mt = mt.filter_cols(hl.is_defined(fam_ht[mt.col_key]))
-    if "adj" not in mt.entry:
-        mt = annotate_adj(mt)
+    if isinstance(t, hl.MatrixTable):
+        t = t.filter_cols(hl.is_defined(fam_ht[t.col_key]))
+    elif isinstance(t, hl.vds.VariantDataset):
+        t = hl.vds.filter_samples(t, fam_ht)
```

Comment:
Should add a note that filtering to trios for the VDS removes some variants, whereas the option for matrix table does not